### DISCUSSION TITLE
Announcement of the Deprecation of PHP language versions less than 7.2.5 in the SDK

### DISCUSSION THREAD

**Discussion by SamRemis** [aws/aws-sdk-php#2638] [Announcements]
On August 15, 2023, in order to continue supporting our customers with tools that are secure and maintainable, AWS will publish a minor version bump of the [AWS SDK for PHP](https://github.com/aws/aws-sdk-php). These new versions will require a PHP 7.2.5+ language version, formally ending our support for versions 7.2.4 and below.

Per the PHP Group, as of November 30th, 2020, no new bug reports, fixes, or changes will be made to PHP 7.2. Recently, many PHP projects and packages have discontinued support PHP versions below 7.2.5. In 2023, we will be joining these deprecations to maintain a secure and up-to-date SDK.

**I use a PHP language level of 7.2.5 or higher, what should I do?**
If you already use a version with continued support, no action is required. You are not impacted by this announcement.

**I use a PHP language level below 7.2.5, what should I do?**
For users below 7.2.5, we recommend updating your PHP language version to a language version supported by the SDK, if possible. Versions of the SDK published mid February 2023 through August 15th 2023 will emit a warning on client construction that your PHP version will be deprecated soon. This can be turned off by setting ‘suppress_php_deprecation_warning’ => true in your client constructor, or globally by setting the `AWS_SUPPRESS_PHP_DEPRECATION_WARNING` environment variable to true.

**What if I can’t upgrade to PHP 7.2.5?**
In the event that you’re unable to migrate to PHP 7.2.5 or higher, previous versions of the AWS SDK for PHP will continue to operate after the deprecation date. Depending on your installation method, you will either stop receiving updates after the deprecation date, or will need to download a release that is compatible with your PHP version.

**Conclusion**
If you’re currently using a version of PHP below 7.2.5 as your runtime for AWS SDK for PHP, we encourage you to upgrade to PHP 7.2.5 or later. Please share your questions, comments, and issues with us on [GitHub](https://github.com/aws/aws-sdk-php).

**Comment by arungowda325** [comment]
We are using PHP 7.0 and the AWS SDK for PHP Version 3. Is the PHP version upgrade mandatory?

**Reply by SamRemis** [reply]
Your old version will continue to work, but you won't be able to install any newer versions that are published after the deprecation date; in other words, if you're content with the features you already have, then you don't need to upgrade until you want any new features that will be published after August 15th, 2023.

**Comment by spaceemotion** [comment]
Will the upgrade also include packages like? [aws/aws-php-sns-message-validator](https://github.com/aws/aws-php-sns-message-validator)
